azkaban-common/src/main/java/azkaban/utils/OsMemoryUtil.java

```diff
@@ -22,16 +23,30 @@
   // This file is used by Linux. It doesn't exist on Mac for example.
   private static final String MEM_INFO_FILE = "/proc/meminfo";
 
-  private static final ImmutableSet<String> MEM_KEYS = ImmutableSet
+  static final ImmutableSet<String> MEM_KEYS = ImmutableSet
       .of("MemFree", "Buffers", "Cached", "SwapFree");
 
+  static final ImmutableSet<String> PHYSICAL_MEM_KEYS = ImmutableSet.of("MemFree");
+
   /**
    * Includes OS cache and free swap.
    *
    * @return the total free memory size of the OS. 0 if there is an error or the OS doesn't support
    * this memory check.
    */
   long getOsTotalFreeMemorySize() {
+    return getAggregatedFreeMemorySize(MEM_KEYS);
+  }
+
+  /**
+   * @return the free physical memory size of the OS. 0 if there is an error or the OS doesn't
+   * support this memory check.
+   */
+  long getOsFreePhysicalMemorySize() {
```

Comment:
What is the reason for checking physical memory instead of total memory?

Reply:
Currently, for `azkaban.polling_criteria.min_free_memory_gb` I'm using a rough approximation of the memory available without pushing the system into swap(it's actually an under estimation), total memory would include swap which I'm trying to avoid.

Based on this [https://unix.stackexchange.com/questions/499649/is-cached-memory-de-facto-free](https://unix.stackexchange.com/questions/499649/is-cached-memory-de-facto-free) I could improve the estimation of free memory by defaulting to field "MemAvailable" whenever is supported. Let me know what you think.

Reply:
Based on your investigation, it seems to me MemAvailable is a better choice. You can do some experiment to verify and improve it later, thanks!

Reply:
sounds good.